pyartcd/pyartcd/pipelines/sign_rhcos_containers.py

```
@@ -0,0 +1,199 @@
+            rhcos_file (str): Path to the rhcos json file
+            arch (str): Architecture (e.g., x86_64, aarch64)
+            signing_env (str | None): Signing environment (stage or prod)
+        """
+        self.runtime = runtime
+        self.rhcos_file = Path(rhcos_file)
+        self.arch = arch
+        self.signing_env = signing_env or "stage"
+        self.logger = self.runtime.logger
+
+        if not self.rhcos_file.exists():
+            raise FileNotFoundError(f"RHCOS file not found: {self.rhcos_file}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)